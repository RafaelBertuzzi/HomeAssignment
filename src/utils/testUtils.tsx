import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { NavigationContainer } from '@react-navigation/native';
import { store } from '@redux/store';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer
    .create(
      <Provider store={store}>
        <NavigationContainer>
          <LayoutProvider>{children}</LayoutProvider>
        </NavigationContainer>
      </Provider>,
    )
    .toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(
    <Provider store={store}>
      <NavigationContainer>
        <LayoutProvider>{children}</LayoutProvider>
      </NavigationContainer>
    </Provider>,
  ).toJSON();
};
