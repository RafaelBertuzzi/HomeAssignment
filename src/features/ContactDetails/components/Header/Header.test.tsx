import { Contact, PostalAddress } from 'react-native-contacts';

import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Header } from '.';

const mockedData = {
  backTitle: '',
  displayName: 'Kate Bell',
  thumbnailPath: '',
  department: 'Marketing',
  isStarred: false,
  note: 'Kate Bell has been working on new designs for our holiday product line. She has some great ideas.',
  prefix: 'Ms.',
  suffix: 'Jr.',
  birthday: { day: 20, month: 1, year: 1978 },
  company: 'Creative Consulting',
  emailAddresses: [{ email: 'kate-bell@mac.com', label: 'work' }],
  familyName: 'Bell',
  givenName: 'Kate',
  hasThumbnail: false,
  imAddresses: [],
  jobTitle: 'Producer',
  middleName: '',
  phoneNumbers: [
    { label: 'mobile', number: '(555) 564-8583' },
    { label: 'main', number: '(415) 555-3695' },
  ],
  postalAddresses: [
    {
      city: 'Hillsborough',
      country: '',
      label: 'work',
      postCode: '94010',
      region: 'CA',
      state: 'CA',
      street: '165 Davis Street',
    },
  ] as PostalAddress[],
  recordID: '177C371E-701D-42F8-A03B-C61CA31627F6',
  urlAddresses: [{ label: 'homepage', url: 'www.icloud.com' }],
} as Contact;

describe('Contact Details Screen Component: Header', () => {
  afterEach(cleanup);

  it('should render', () => {
    const json = rendererWithProviders(<Header contact={mockedData} />);
    expect(json).toMatchSnapshot();
  });
});
