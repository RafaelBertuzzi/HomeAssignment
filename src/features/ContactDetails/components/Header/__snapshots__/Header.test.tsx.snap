// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Details Screen Component: Header should render 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={
      {
        "position": "fixed",
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginRight": 16,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 10,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={{}}
            style={{}}
          >
            <Image
              accessibilityLabel="left-arrow"
              alt="left-arrow"
              dataSet={{}}
              onError={[Function]}
              source={
                {
                  "testUri": "../../../src/assets/icons/left-arrow.png",
                }
              }
              style={
                {
                  "height": 24,
                  "maxWidth": "100%",
                  "width": 24,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 10,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={{}}
            style={{}}
          >
            <Image
              accessibilityLabel="star-outline"
              alt="star-outline"
              dataSet={{}}
              onError={[Function]}
              source={
                {
                  "testUri": "../../../src/assets/icons/star-outline.png",
                }
              }
              style={
                {
                  "height": 24,
                  "maxWidth": "100%",
                  "width": 24,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      aria-orientation="horizontal"
      dataSet={{}}
      style={
        {
          "backgroundColor": "#d4d4d4",
          "height": 1,
          "width": "100%",
        }
      }
      thickness="1"
    />
  </View>
</RNCSafeAreaProvider>
`;
